apps/api/src/app/change/usecases/promote-layout-change/promote-layout-change.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@nestjs/common';
+import { LayoutEntity, LayoutRepository } from '@novu/dal';
+
+import { PromoteTypeChangeCommand } from '../promote-type-change.command';
+
+@Injectable()
+export class PromoteLayoutChange {
+  constructor(private layoutRepository: LayoutRepository) {}
+
+  async execute(command: PromoteTypeChangeCommand) {
+    const item = await this.layoutRepository.findOne({
+      _environmentId: LayoutRepository.convertStringToObjectId(command.environmentId),
+      _id: command.item._id,
+    });
+
+    const newItem = command.item as LayoutEntity;
+
+    if (!item) {
+      const layoutEntity = {
+        ...newItem,
```

Comment:
Same here, if there isn't an item that has already been promoted from this item id, one should be created with:
`_parentId:  newItem._id`